hermeto/core/package_managers/pip/rust.py

```diff
@@ -24,70 +23,69 @@ def _has_rust_build_deps(raw_build_dependencies: list[str]) -> bool:
+            # pybuild-deps parser has some known edge-cases for older packages relying on setup.py
+            log.error("Unable to parse build dependencies for %s.", extracted_dir.name)
             continue
+
         if _has_rust_build_deps(build_dependencies):
             return True
+
     return False
 
 
 def filter_packages_with_rust_code(packages: list[dict[str, Any]]) -> list[CargoPackageInput]:
     """Filter packages that contain Rust code from a list of pip packages."""
     packages_containing_rust_code = []
-    tar_packages = [p for p in packages if tarfile.is_tarfile(p.get("path", ""))]
-    for p in tar_packages:
-        fname = p.get("path", "")
-        # File name and package name may differ e.g. when there is a hyphen in
-        # package name it might be replaced by an underscore in a file name.
-        pname = Path(Path(fname).name)
-        while pname.suffix in (".tar", ".gz", ".tgz"):
-            pname = pname.with_suffix("")
-        tf = tarfile.open(fname)
-        toplevel_cargo = f"{pname}/Cargo.toml"
-        try:
-            tf.getmember(toplevel_cargo)
-            rust_root = pname
-        except KeyError:
-            # only skip if no Cargo.toml is present in the package
-            if not any([fname.endswith("Cargo.toml") for fname in tf.getnames()]):
-                continue
-            # considering it has a Cargo.toml, let's check if it depends on the typical toolchain
-            # for python+rust to rule out false positives
-            if not _depends_on_rust(tf):
-                continue
-            # find the top-most Cargo.toml in the package - that's not necessarily the most accurate
-            # solution, but this heuristic has proven to work on the most popular python packages
-            # that have rust dependencies; if this stops working, then we would probably need to check
-            # pyproject toml config section for maturin or setuptools-rust...
-            # More info on this issue in the design doc
-            # https://github.com/hermetoproject/hermeto/blob/e5fa5c0fcd0dff62cf02be5b0d219e04c1ea440c/docs/design/cargo-support.md#L806
-            cargo_manifests = [name for name in tf.getnames() if name.endswith("Cargo.toml")]
-            rust_root = Path(sorted(cargo_manifests, key=len)[0]).parent
-
-        tf.extractall(path=Path(str(tf.name)).parent, filter="data")
-        packages_containing_rust_code.append(CargoPackageInput(type="cargo", path=rust_root))
+
+    for p in packages:
+        package_path = p.get("path", "")
+        if not package_path:
+            continue
+
+        filename = Path(Path(package_path).name)
+        while filename.suffix in (".zip", ".tar", ".gz", ".tgz"):
+            filename = filename.with_suffix("")
+
+        pip_deps_dir = Path(package_path).parent
+        extract_dir = pip_deps_dir / filename
+        shutil.unpack_archive(package_path, extract_dir=extract_dir)
```

Comment:
Do we need to worry that we're extracting archives that are untrusted from our perspective?